ddtrace/tracer.py

```diff
@@ -446,7 +446,7 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
         self._check_new_process()
 
         # update set of services handled by tracer
-        if service and service not in self._services:
+        if service and service not in self._services and self._is_span_internal(span):
             self._services.add(service)
```

Comment:
should this just be moved up to L419?

```
if self._runtime_worker and self._is_span_internal(span):
    if service and service not in self._services:
        self._services.add(service)
```

not sure what benefits we have for having it down here, but just a thought, not a required change.